lib/segment/src/index/query_optimization/rescore_formula/formula_scorer.rs

```diff
@@ -144,7 +144,9 @@ impl FormulaScorer<'_> {
                 let right = self.eval_expression(right, point_id)?;
                 // avoid division by zero
                 if right == 0.0 {
-                    Ok(*by_zero_default)
+                    by_zero_default.ok_or_else(|| OperationError::NonFiniteNumber {
+                        expression: format!("{left}/{right}"),
```

Comment:
Would it make sense to include the result, being NaN or Inf?

```suggestion
                        expression: format!("{left}/{right}={}", left / right),
```

Reply:
Probably yes, so that it makes it easier to choose a sane default